### PR TITLE
RWA-883: Permissions held by task

### PR DESCRIPTION
**Pull request by alexbaeza**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java [hunk 1]
```diff
@@ -122,6 +122,20 @@ private DslPart createResponseForGetTask() throws JsonProcessingException {
                         .stringType("case_name", "Bob Smith")
                         .booleanType("warnings", false)
                         .stringType("work_type", "hearing_work")
+                    // Note: Array verification cannot be done since unorderedArray is introduced in v4 Specification
+                    // See: RWA-900
```
Comment (satvinder-hullait) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java:126]: are we not better off just removing this code and have the comment in the commit history?

**Review by satvinder-hullait — APPROVED** [review]

**Review by dharmendrak — DISMISSED** [review]
LGTM

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java (hunk 1 above)
Comment (alexbaeza) [reply src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java]: yes you're right I'll just remove it

File: src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java [hunk 2]
```diff
@@ -219,7 +156,17 @@ private DslPart createResponseForGetTaskWithWarnings() throws JsonProcessingExce
                                 .stringType("warningText", "Text1")
                             )
                         )
-                        .stringType("case_management_category", "Some Case Management Category")
```
Comment (BenNzewi) [inline src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java:222]: workType mapping is no longer here? -> .stringType("work_type_id", "hearing_work")

Comment (alexbaeza) [reply src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/consumer/wa/TaskManagementGetTasksBySearchCriteriaConsumerTest.java:222]: solved thanks

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/repository/TaskResourceRepository.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
     Optional<TaskResource> getByTaskId(String id);
 
-    @Modifying
+    @ModifyingTaskManagementGetTasksBySearchCriteriaConsumerTest.java
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/repository/TaskResourceRepository.java:30]: oops :)

**Review by serdaroguzhanakin — DISMISSED** [review]

**Pull request merged**